[Amadeus] ip

ama-chi opened the pull request.

ama-chi committed:
Level 1

ama-chi committed:
Level-1

ama-chi committed:
Level-2

ama-chi committed:
Level-3

ama-chi committed:
Level-3

ama-chi committed:
Level-4

ama-chi committed:
Level-4

ama-chi committed:
Level-5

ama-chi committed:
A-TextUiTesting

ama-chi committed:
Level-6

ama-chi committed:
Fixed branch name

ama-chi committed:
Add DateTime parser

ama-chi committed:
Merge changes from branch-Level-7

ama-chi committed:
Merge branch 'branch-Level-8'

ama-chi committed:
Delete unnecesary files

ama-chi committed:
Integrating Gradle

ama-chi committed:
Add more OOP

ama-chi committed:
Add package

ama-chi committed:
Add Junit testing

ama-chi committed:
Add javadocs

ama-chi committed:
Improve coding standard

ama-chi committed:
Add find

ama-chi committed:
Improve Coding Standard

ama-chi committed:
Add Find

kangzongxian commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,174 @@
+ */
+public class Duke {
+
+    /**
```
Good to list down the description of the attributes!

kangzongxian commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,174 @@
+     */
+    private void executeCommand(Commands command, String input) {
+        if (command == Commands.LIST) {
+            System.out.println("    ____________________________________________________________\n");
```
For consistency sake, I think it might be better to abstract out this line break into a variable, so that you can just use it!

kangzongxian commented on src/main/java/Duke/Task.java:
```diff
@@ -0,0 +1,108 @@
+    public void mark() {
+        completed = true;
+        System.out.printf(
+                "    ____________________________________________________________\n" +
```
Likewise here, I think it would be good to abstract out the line break into a variable as well!

kangzongxian commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,174 @@
+package Duke;
```
Overall I find everything looks really good, with JavaDocs well written and good testing, nice!

kangzongxian commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,174 @@
+            }
+            System.out.println("    ____________________________________________________________\n");
+
+        } else if (command == Commands.MARK) {
```
I think that there may be a lot of different cases going on here, so it may be a good idea to leave a comment on what case each block is handling!

ama-chi committed:
Add Level-10

kynapy commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,174 @@
+                if (newTask != null) {
+                    storage.writeToFile(newTask);
+                }
+            } catch (DukeException.EmptyTaskException | DukeException.UnkownCommandException | DukeException.InvalidParameterException error) {
```
Line might be too long, could try to wrap the last exception into the next line!

kynapy commented on src/main/java/Duke/Task.java:
```diff
@@ -0,0 +1,108 @@
+    /**
+     * Creates a String representation of Duke.Task.
+     */
+    public String toString() {
```
Good to add in @ Override when overriding methods!

kynapy reviewed (commented):
LGTM, just some nits such as long lines of code that could be wrapped to the next lines and the repeated long underscores could be stored into a variable for more concise code. Cheers 😄

ama-chi committed:
Image retrieval: may return null.

Program may not run as intended if file is null.

Add assert to check for this.

Assert helps to check if the image is found.

Assert is done before dialogContainer is called to process the images so that it is not used unnecesarily.

Changes were made to only Duke.java and MainWindow.java.

ama-chi committed:
Newly written code not updated with code quality.
Ammendments to make code easier to read and follow.
Lengthy under execute abstracted to a separate "Execute" class.
This is done to make execute function under Duke less messy and more readable.
Smaller style changes are made throughout the project as well.

ama-chi committed:
Merge pull request #1 from ama-chi/branch-Assertions

Add assertions

ama-chi committed:
Merge update Master

ama-chi committed:
Merge pull request #2 from ama-chi/branch-CodeQuality

Improve code quality

ama-chi committed:
Add FixedDurationTask

ama-chi committed:
Add UG

ama-chi committed:
Release
